distributed/scheduler.py

```diff
@@ -7486,6 +7495,17 @@ def decide_worker(
         candidates = set(all_workers)
     else:
         candidates = {wws for dts in deps for wws in dts._who_has}
+        # Add some random workers to into `candidates`, starting with idle ones
+        # TODO shuffle to prevent hotspots?
+        candidates.update(idle_workers[:N_RANDOM_WORKERS])
```

Comment:
We *may* want to randomly select here, I'm not sure.  It probably doesn't matter that much because we'll naturally cycle the chosen ones out of the group.

Reply:
We wouldn't shuffle, to be clear, we would index.

Also, maybe 5 or 10 workers is enough?

Reply:
> It probably doesn't matter that much because we'll naturally cycle the chosen ones out of the group.

That's what I figured. I was more thinking about a regression on #4637 for the "ish" tasks of root-ish that wouldn't go down the current round-robin code path, since they do have dependencies.